### PR TITLE
Make the world initialization slightly fast part 2

### PR DESCRIPTION
**Pull request by eswordthecat**

### CODE REVIEW

File: code/modules/research/rdmachines.dm [hunk 1]
```diff
@@ -48,20 +49,28 @@ var/global/list/rnd_machines = list()
 	src.disable_wire = pick(w)
 	w -= src.disable_wire
 
+	if (ticker)
```
Comment (d3athrow) [inline code/modules/research/rdmachines.dm]: so to make it initialize you have to decon and rebuild it? all this stuff's new is called before the ticker is made

Comment (eswordthecat) [inline code/modules/research/rdmachines.dm]: no

```
datum/controller/game_controller/proc/setup_objects()
	world << "<span class='danger'>Initializing objects</span>"
	sleep(-1)
	//var/last_init_type = null
	for(var/atom/movable/object in world)
		//if(last_init_type != object.type)
		//	testing("Initializing [object.type]")
		//	last_init_type = object.type
		object.initialize()
```

**Comment by eswordthecat** [comment]
merge when

File: code/modules/research/rdmachines.dm [hunk 2]
```diff
@@ -25,15 +25,16 @@ var/global/list/rnd_machines = list()
...
+	var/list/datum/materials/materials
 	var/max_material_storage = 0
 	var/list/allowed_materials[0]
 
 	var/research_flags //see setup.dm for details of these
 
-/obj/machinery/r_n_d/New()
-	rnd_machines |= src
-	..()
+/obj/machinery/r_n_d/New(loc)
```
Comment (ComicIronic) [inline code/modules/research/rdmachines.dm]: There was no reason to specify the args.
Why would you

**Pull request closed without merging**